graspologic/layouts/__main__.py

```diff
@@ -17,13 +17,24 @@ def _graph_from_file(
     path: str,
     skip_header: bool = False,
 ) -> nx.Graph:
+    logger = logging.getLogger("graspologic.layouts")
     graph = nx.Graph()
     with open(path, "r") as edge_io:
         if skip_header is True:
             next(edge_io)
+        first = True
         for line in edge_io:
-            source, target, weight = line.strip().split(",")
-            weight = float(weight)
+            split_vals = line.strip().split(",")
+            if len(split_vals) == 3:
+                source, target, weight = split_vals
+                weight = float(weight)
+            elif len(split_vals) == 2:
+                if first:
+                    logger.warn("No weights found in edge list, using 1.0")
+                    first = False
+                source, target = split_vals[:2]
+                weight = 1.0
+            #else drop it because it is malformed
```

Comment:
I'd assert you should throw an error and stop processing, we need to know early on if we have malformed data not just keep processing

Reply:
It could be the case that we just have a non-weighted edge list. In that scenario I think the one time warning it appropriate. Maybe I should throw an exception where the #else is. Should I change that case?

Reply:
I think it should be:

```python
else:
   if len(line.strip()) == 0:
     # do nothing
   else:
     raise IOError(f"Expected 2 or 3 columns in {path}, no more or less")
```

Reply:
I think this is a good idea. I just committed it this way.